libkbfs/block_journal.go

```diff
@@ -362,22 +382,19 @@ func (j *bserverTlfJournal) getAll() (
 	return res, nil
 }
 
-func (j *bserverTlfJournal) putData(
+func (j *blockJournal) putData(
 	id BlockID, context BlockContext, buf []byte,
 	serverHalf BlockCryptKeyServerHalf) error {
 	err := validateBlockServerPut(j.crypto, id, context, buf)
 	if err != nil {
 		return err
 	}
 
-	j.lock.Lock()
-	defer j.lock.Unlock()
-
 	if j.isShutdown {
-		return errBserverTlfJournalShutdown
+		return errBlockJournalShutdown
 	}
 
-	_, existingServerHalf, err := j.getDataLocked(id, context)
+	_, existingServerHalf, err := j.getDataWithContext(id, context)
```

Comment:
Ah, that's on purpose. Since we skip the write if there is existing data below, I wanted to make sure said existing data was valid. Added comment.